pkg/metrics/logging_hook.go

```diff
@@ -4,19 +4,29 @@
 package metrics
 
 import (
+	"context"
 	"fmt"
-	"reflect"
+	"io"
+	"log/slog"
+	"maps"
+	"os"
+	"strconv"
 	"sync"
 	"sync/atomic"
 
-	"github.com/sirupsen/logrus"
-
 	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
+var slogDupAttrDetection = false
+
+func init() {
+	// Detector to check if we have duplicate attributes in slog logging hook.
+	slogDupAttrDetection, _ = strconv.ParseBool(os.Getenv("CILIUM_INVALID_METRIC_VALUE_DETECTOR"))
```

Comment:
nit: Looks like this feature leaked into the second commit, but the third commit seems to be the one intended to introduce the feature.